doozer/doozerlib/util.py

```diff
@@ -540,20 +540,57 @@ def get_release_name_for_assembly(group_name: str, releases_config: Model, assem
     return get_release_name(assembly_type, group_name, assembly_name, patch_version)
 
 
-def oc_image_info(pullspec: str, *options):
+def oc_image_info(
+        pullspec: str,
+        *options,
+        registry_config: Optional[str] = None,
+        registry_username: Optional[str] = None,
+        registry_password: Optional[str] = None
+) -> Union[Dict, List]:
     """
     Returns a Dict of the parsed JSON output of `oc image info` for the specified
     pullspec. Use oc_image_info__caching if you do not believe the image will change
     during the course of doozer's execution.
 
     :param pullspec: e.g. registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-vsphere-problem-detector-rhel9:v4.19.0-202501230108.p0.gcbd8539.assembly.stream.el9
     :param options: list of extra args to use with oc, e.g. '--show-multiarch', '--filter-by-os=linux/amd64'
+    :param registry_config: The path to the registry config file.
+    :param registry_username: The username to authenticate with the registry.
+    :param registry_password: The password to authenticate with the registry.
     """
 
-    cmd = ['oc', 'image', 'info', '-o', 'json', pullspec]
-    cmd.extend(options)
-    out, _ = exectools.cmd_assert(cmd, retries=3)
-    return json.loads(out)
+    def run_oc(auth_file: Optional[str]):
+        cmd = ['oc', 'image', 'info', '-o', 'json', pullspec]
+        cmd.extend(options)
+        if auth_file:
+            cmd.extend([f'--registry-config={auth_file}'])
+        out, _ = exectools.cmd_assert(cmd, retries=3)
+        return json.loads(out)
+
+    # If neither a username nor a password were provided, assume an auth file is being used or no auth at all
+    if not registry_username and not registry_password:
+        return run_oc(auth_file=registry_config)
+
+    # Otherwise, a password is always needed for basic auth
+    assert registry_password, 'registry_password must be provided if auth is needed.'
```

Comment:
So we need either auth_file or password to use this func. I would say it would  be fine to run if none is present, as we were doing before?

Reply:
if you don't provide anything, oc runs without auth options. If you provide a `registry_config`, it must point to a file where auth credentials are stored. Otherwise, you must provide username and password to be used with basic auth (which will be used to create a temporary registry config file)